app/main.py

```diff
@@ -1,34 +1,110 @@
+from __future__ import annotations
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: int, end: int, is_drowned: bool = False) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_decks()
+
+    def create_decks(self) -> list[Deck]:
+        decks = []
+        if self.start[0] == self.end[0]:
+            for col in range(self.start[1], self.end[1] + 1):
+                decks.append(Deck(self.start[0], col))
+        elif self.start[1] == self.end[1]:
+            for row in range(self.start[0], self.end[0] + 1):
+                decks.append(Deck(row, self.start[1]))
```

Comment:
The `start` and `end` variables are expected to be tuples, but they are used as if they are iterable with indices. Ensure that `start` and `end` are tuples of integers, e.g., `(row, column)`.